src/transaction/log_manager.c

```diff
@@ -10214,6 +10418,3430 @@ logtb_tran_update_stats_online_index_rb (THREAD_ENTRY * thread_p, void *data, vo
+		  {
+		    error_code = ER_OUT_OF_VIRTUAL_MEMORY;
+		    goto end;
+		  }
+
+		memcpy (undo_recdes->data, (char *) undo_data + sizeof (undo_recdes->type), undo_recdes->length);
+	      }
+
+	    break;
+	  }
+
+	case LOG_REDO_DATA:
+	  {
+	    LOG_REC_REDO *redo = NULL;
+	    LOG_RCVINDEX rcvindex;
+
+	    redo = (LOG_REC_REDO *) (log_page_p->area + process_lsa.offset);
+	    rcvindex = redo->data.rcvindex;
+
+	    if (rcvindex == RVOVF_PAGE_UPDATE)
+	      {
+		/* GET OVF UNDO IMAGE */
+		if ((error_code =
+		     cdc_get_overflow_recdes (thread_p, log_page_p, undo_recdes, *undo_lsa, rcvindex,
+					      false)) != NO_ERROR)
+		  {
+		    goto end;
+		  }
+	      }
+
+	    break;
+	  }
+
+	default:
+	  break;
+	}
+
+      if (is_undo_alloced)
+	{
+	  free_and_init (undo_data);
+	  is_undo_alloced = false;
+	}
+    }
+
+/* Get REDO RECDES from redo lsa */
+  if (redo_lsa != NULL)
+    {
+      tmpbuf_index = redo_lsa->pageid % 2;
+
+      if (LSA_ISNULL (&process_lsa) && process_lsa.pageid == redo_lsa->pageid)
+	{
+	  /* if undo_lsa != NULL and current log_page_p can be reusable */
+	  assert (log_page_p != NULL);
+	}
+      else
+	{
+	  if (temp_logbuf[tmpbuf_index].pageid == redo_lsa->pageid)
+	    {
+	      log_page_p = temp_logbuf[tmpbuf_index].log_page_p;
+	    }
+	  else
+	    {
+	      log_page_p = (LOG_PAGE *) PTR_ALIGN (log_pgbuf, MAX_ALIGNMENT);
+
+	      if (logpb_fetch_page (thread_p, redo_lsa, LOG_CS_SAFE_READER, log_page_p) != NO_ERROR)
+		{
+		  error_code = ER_FAILED;
+		  goto end;
+		}
+	    }
+	}
+
+      LSA_COPY (&process_lsa, redo_lsa);
+
+      log_rec_hdr = LOG_GET_LOG_RECORD_HEADER (log_page_p, &process_lsa);
+
+      LOG_READ_ADD_ALIGN (thread_p, sizeof (*log_rec_hdr), &process_lsa, log_page_p);
+
+      switch (log_rec_hdr->type)
+	{
+	case LOG_MVCC_DIFF_UNDOREDO_DATA:
+	case LOG_MVCC_UNDOREDO_DATA:
+	  {
+	    LOG_REC_MVCC_UNDOREDO *mvcc_undoredo = NULL;
+	    LOG_RCVINDEX rcvindex;
+
+	    LOG_READ_ADVANCE_WHEN_DOESNT_FIT (thread_p, sizeof (*mvcc_undoredo), &process_lsa, log_page_p);
+
+	    mvcc_undoredo = (LOG_REC_MVCC_UNDOREDO *) (log_page_p->area + process_lsa.offset);
+	    rcvindex = mvcc_undoredo->undoredo.data.rcvindex;
+	    redo_length = mvcc_undoredo->undoredo.rlength;
+	    undo_length = mvcc_undoredo->undoredo.ulength;
+
+	    if (LOG_IS_DIFF_UNDOREDO_TYPE (log_rec_hdr->type) == true)
+	      {
+		is_diff = true;
+	      }
+
+	    LOG_READ_ADD_ALIGN (thread_p, sizeof (*mvcc_undoredo), &process_lsa, log_page_p);
+
+	    if (rcvindex == RVHF_MVCC_INSERT)
+	      {
+		MVCC_REC_HEADER mvcc_rec_header;	/*To clear mvcc rec header for MVCC INSERT , because RECDES in log record for RVHF_MVCC_INSERT does not contain */
+		char *tmp_ptr;
+
+		if (ZIP_CHECK (redo_length))
+		  {
+		    redo_length = (int) GET_ZIP_LEN (redo_length);
+		    is_zipped_redo = true;
+		  }
+
+		if (process_lsa.offset + redo_length < (int) LOGAREA_SIZE)
+		  {
+		    redo_data = (char *) (log_page_p->area + process_lsa.offset);
```

Comment:
Is it OK no to skip the undo part?

Reply:
Even if the undo part can be skipped, how about checking the checking and skipping. This code assumes the undo part of RVHF_MVCC_INSERT is 0, but it may be changed later.